sw/device/lib/dif/dif_uart.c

```diff
@@ -105,8 +106,20 @@ static dif_uart_config_result_t uart_configure(
     return kDifUartConfigBadConfig;
   }
 
-  // Calculation formula: NCO = 2^20 * baud / fclk.
-  uint64_t nco = ((uint64_t)config->baudrate << 20) / config->clk_freq_hz;
+  // Calculation formula: NCO = 16 * 2^nco_width * baud / fclk.
+
+  // Compute NCO register bit width
+  uint32_t nco_width = 0;
+
+  for (int i = 0; i < 32; i++) {
+    nco_width += (UART_CTRL_NCO_MASK >> i) & 1;
+  }
+
+  _Static_assert((UART_CTRL_NCO_MASK >> 28) == 0,
+                 "NCO bit width exceeds 28 bits.");
+
+  uint64_t nco =
```

Comment:
Could we please drop a comment here what `4` means in this case?

Reply:
Added.